src/installer.ts

```diff
@@ -273,19 +275,21 @@ export class Installer extends EventEmitter {
     return promise;
   }
 
-  /** the currently-installing version, if any */
-  private installing: string | undefined;
+  /** map of version string to the status of its installing state */
+  private installing = new Map<string, boolean>();
```

Comment:
question: if the value in the map is just a boolean value, can we instead just have a `Set()` and add/remove items when needed?

Reply:
Yup, using `Set` here is a much cleaner approach :)